documentation/modules/deploying/proc-deploy-migrate-kraft.adoc

```diff
@@ -0,0 +1,256 @@
+metadata:
+  name: controller
+  labels:
+    strimzi.io/cluster: my-cluster
+spec:
+  replicas: 3
+  roles:
+    - controller
+  storage:
+    type: jbod
+    volumes:
+      - id: 0
+        type: persistent-claim
+        size: 20Gi
+        deleteClaim: false
+    resources:
+      requests:
+        memory: 64Gi
+        cpu: "8"
+      limits:
+        memory: 64Gi
+        cpu: "12"    
+----
++
+NOTE: For the migration, you cannot use a node pool of nodes that share the broker and controller roles.
+
+. Apply the new `KafkaNodePool` resource to create the controllers.
++
+Errors related to using controllers in a ZooKeeper-based environment are expected.
+
+. Enable KRaft migration in the `Kafka` resource by setting the `strimzi.io/kraft` annotation to `migration`:
++
+[source,shell]
+----
+kubectl annotate kafka my-cluster strimzi.io/kraft: migration
+----
++
+.Enabling KRaft migration
+[source,yaml,subs="+attributes"]
+----
+apiVersion: {KafkaApiVersion}
+kind: Kafka
+metadata:
+  name: my-cluster
+  namespace: my-project
+  annotations:
+    strimzi.io/kraft: migration 
+# ...
+----
++
+Applying the annotation to the `Kafka` resource configuration starts the migration.
+
+. Check the controllers have started and the brokers have rolled:
++
+[source,shell]
+----
+kubectl get pods -n my-project
+----
++
+.Output shows nodes in broker and controller node pools
+[source,shell]
+----
+NAME                     READY  STATUS   RESTARTS
+my-cluster-kafka-0       1/1    Running  0
+my-cluster-kafka-1       1/1    Running  0
+my-cluster-kafka-2       1/1    Running  0
+my-cluster-controller-3  1/1    Running  0
+my-cluster-controller-4  1/1    Running  0
+my-cluster-controller-5  1/1    Running  0
+# ...
+----
++
+Here, the brokers have the name `my-cluster-kafka` as they were contained in a node pool named `kafka` when migrating the cluster to use node pools.
+
+. Check the status of the migration:
++
+[source,shell]
+----
+kubectl get kafka my-cluster -n my-project -w
+----
++
+.Updates to the metadata state
+[source,shell]
+----
+NAME        ...  METADATA STATE
+my-cluster  ...  Zookeeper
+my-cluster  ...  KRaftMigration
+my-cluster  ...  KRaftDualWriting
+my-cluster  ...  KRaftPostMigration
+----
++
+`METADATA STATE` shows the mechanism used to manage Kafka metadata and coordinate operations.
+At the start of the migration this is `ZooKeeper`.
++
+--
+* `ZooKeeper` is the initial state when metadata is only stored in ZooKeeper.
+* `KRaftMigration` is the state when the migration is in progress and brokers are rolled to register with the controllers.
+The migration can take some time at this point depending on the number of topics and partitions in the cluster. 
+* `KRaftDualWriting` is the state when the Kafka cluster is working as a KRaft cluster, 
+but controllers are managing metadata in Kafka and ZooKeeper. 
+Brokers are rolled a second time to remove the ZooKeeper configuration and migration annotation from the `Kafka` resource.
```

Comment:
`and migration annotation from the `Kafka` resource.` we are not removing stuff from the Kafka resource.
Maybe you meant the ZooKeeper migration flag from the broker configuration.

Reply:
It's not an annotation. It's a configuration applied/removed by the operator on the broker when you migrate/end migration. You are mentioning ZooKeepeer connection here, it's not an annotation as well. The ZooKeeper migration flag comes along with such connection info.

Reply:
I guess it's this `zookeeper.metadata.migration.enable`, but maybe we don't need to mention that. Is it added at `KRaftMigration`?  And in which state is the ZooKeeper configuration and related resources removed?

Reply:
Yes that flag exactly. It's added when moving to KRftMigration and removed when moving out of KRaftDualWriting. Anyway I am not saying we should mention it but here we have a wrong sentence about "remove the annotation on the Kafka custom resource" which is wrong. We are not removing any annotation.

Reply:
Thanks. I've mentioned when the flag is added and when its removed. I think it's useful.